### PR TITLE
PRIME-2320 Oral Health Professional v4

### PR DESCRIPTION
**Pull request by bergomi02**

### CODE REVIEW

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/CollegeLicenseConfiguration.cs [hunk 1]
```diff
@@ -45,7 +45,6 @@ public override IEnumerable<CollegeLicense> SeedData
                     new CollegeLicense { CollegeCode = 1, LicenseCode = 89 },
                     new CollegeLicense { CollegeCode = 1, LicenseCode = 90 },
                     new CollegeLicense { CollegeCode = 1, LicenseCode = 91 },
-                    new CollegeLicense { CollegeCode = 1, LicenseCode = 92 },
```
Comment (neophyte57) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/CollegeLicenseConfiguration.cs:48]: Eliminating this (for Certified Physician Assistant) doesn't seem correct?

Comment (bergomi02) [reply prime-dotnet-webapi/Configuration/Database/Seeded Tables/CollegeLicenseConfiguration.cs]: Good catch! It is fixed and the database has been rebuilt.

File: prime-dotnet-webapi/prime-dotnet-webapi.sln [hunk 2]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
```
Comment (neophyte57) [inline prime-dotnet-webapi/prime-dotnet-webapi.sln:2]: Is this file necessary?

**Comment by neophyte57** [comment]
1. For existing data (enrolled with College of Dental Surgeons of BC), so far so good:

image

2. But when try to edit licence, get:

image

**Comment by neophyte57** [comment]
Student Dentist is defaulting to RU TOA:
image

I think NIMR and LPC values need to be adjusted from what is in spreadsheet.  Let's talk.

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs [hunk 3]
```diff
@@ -325,7 +326,42 @@ public override IEnumerable<LicenseDetail> SeedData
...
+                    new LicenseDetail { Id =  295, LicenseCode = 111, Prefix = "95", Manual = true, Validate = false, LicensedToProvideCare = true, NamedInImReg = true, PrescriberIdType = null, AllowRequestRemoteAccess = false, EffectiveDate = OralProfessionalDec112023, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE },
+                    new LicenseDetail { Id =  296, LicenseCode = 112, Prefix = "95", Manual = true, Validate = true, LicensedToProvideCare = false, NamedInImReg = true, PrescriberIdType = null, AllowRequestRemoteAccess = false, EffectiveDate = OralProfessionalDec112023, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE },
+                    new LicenseDetail { Id =  297, LicenseCode = 113, Prefix = "95", Manual = true, Validate = true, LicensedToProvideCare = true, NamedInImReg = true, PrescriberIdType = null, AllowRequestRemoteAccess = false, EffectiveDate = OralProfessionalDec112023, CreatedTimeStamp = SEEDING_DATE, UpdatedTimeStamp = SEEDING_DATE },
```
Comment (neophyte57) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs:356]: `Manual = true` should be `false`

Comment (bergomi02) [reply prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseDetailConfiguration.cs]: This is fixed.

**Comment by bergomi02** [comment]
> 1. For existing data (enrolled with College of Dental Surgeons of BC), so far so good:
> 
> image
> 
> 2. But when try to edit licence, get:
> 
> image

I believe it is fixed.

**Comment by bergomi02** [comment]
> Student Dentist is defaulting to RU TOA: image
> 
> I think NIMR and LPC values need to be adjusted from what is in spreadsheet. Let's talk.

This is interesting. After I tested couple cases, it turns out setting LPC to false will set the TOA to none. Only setting NIMR to false will set the TOA to OBO. So, the NIMR of both license types are set to false with comment.

File: prime-dotnet-webapi/prime-dotnet-webapi.sln (hunk 2 above)
Comment (bergomi02) [reply prime-dotnet-webapi/prime-dotnet-webapi.sln:2]: Not necessary, but VS code keep adding it back for some reason. I just let it through since it won't hurt anything, then I don't need to remove it every build.

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs [hunk 4]
```diff
@@ -109,6 +109,47 @@ public override IEnumerable<License> SeedData
...
+                    new License { Code = 106, Weight = 100, Name = "Full Dentist" },
+                    new License { Code = 107, Weight = 102, Name = "Limited (Academic) Dentist" },
+                    new License { Code = 108, Weight = 103, Name = "Limited (Armed Services or Government) Dentist" },
+                    new License { Code = 109, Weight = 104, Name = "Limited (Education & Volunteer) Dentist" },
+                    new License { Code = 110, Weight = 105, Name = "Limited (Restricted-to-Specialty) Dentist" },
+                    new License { Code = 111, Weight = 106, Name = "Student Dentist" },
+                    new License { Code = 112, Weight = 107, Name = "Non-Practising Dentist" },
+                    new License { Code = 113, Weight = 108, Name = "Temporary Dentist" },
+
+                    // Denturist
```
Comment (neophyte57) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs:145]: Please check ordering of licence classes versus https://oralhealthbc.ca/register-renew/registrant-classifications/denturist/

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs [hunk 5]
```diff
@@ -109,6 +109,47 @@ public override IEnumerable<License> SeedData
...
+                    new License { Code = 95, Weight = 123, Name = "Non-Practising Certified Dental Assistant" },
+                    new License { Code = 96, Weight = 124, Name = "Temporary Certified Dental Assistant" },
+
+                    // Dental Hygienist
+                    new License { Code = 97, Weight = 131, Name = "Registered Dental Hygienist" },
+                    new License { Code = 98, Weight = 132, Name = "Dental Hygiene Practitioner" },
+                    new License { Code = 99, Weight = 133, Name = "Non-Practising Dental Hygienist" },
+                    new License { Code = 100, Weight = 134, Name = "Temporary Dental Hygienist" },
+
+                    // Dental Technician
```
Comment (neophyte57) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs:126]: Please check ordering of licence classes versus https://oralhealthbc.ca/register-renew/registrant-classifications/dental-technician/

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs [hunk 6]
```diff
@@ -109,6 +109,47 @@ public override IEnumerable<License> SeedData
                     new License { Code = 85, Weight = 4, Name = "Non-practicing registration" },
                     new License { Code = 86, Weight = 4, Name = "Temporary registration" },
 
+
+                    //College of Oral Health Professionals
+                    // Dental Assistant
```
Comment (neophyte57) [inline prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs:114]: Please check ordering of licence classes versus https://oralhealthbc.ca/register-renew/registrant-classifications/certified-dental-assistant/

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs (hunk 4 above)
Comment (bergomi02) [reply prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs:145]: The last 3 licence classes have been re-ordered from Leanne in the excel spreadsheet. It is different that the order in web site.

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs (hunk 5 above)
Comment (bergomi02) [reply prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs:126]: The last 3 licence classes have been re-ordered from Leanne in the excel spreadsheet. It is different that the order in web site.

File: prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs (hunk 6 above)
Comment (bergomi02) [reply prime-dotnet-webapi/Configuration/Database/Seeded Tables/LicenseConfiguration.cs:114]: The last 3 licence classes have been re-ordered from Leanne in the excel spreadsheet. It is different that the order in web site.

File: prime-angular-frontend/src/app/modules/adjudication/pages/license-classes-maintenance-page/license-classes-maintenance-page.component.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+.discontinued {
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/pages/license-classes-maintenance-page/license-classes-maintenance-page.component.scss:1]: Is this used?

Comment (bergomi02) [reply prime-angular-frontend/src/app/modules/adjudication/pages/license-classes-maintenance-page/license-classes-maintenance-page.component.scss]: the code is removed.

File: prime-angular-frontend/src/app/modules/adjudication/pages/license-classes-maintenance-page/license-classes-maintenance-page.component.scss [hunk 8]
```diff
@@ -0,0 +1 @@
+
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/modules/adjudication/pages/license-classes-maintenance-page/license-classes-maintenance-page.component.scss:1]: Still a change snuck in  🫥

File: prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.ts [hunk 9]
```diff
@@ -382,4 +412,17 @@ export class CollegeCertificationFormComponent implements OnInit {
   private filterPractices(collegeCode: number): PracticeConfig[] {
     return this.practices.filter(p => p.collegePractices.map(cl => cl.collegeCode).includes(collegeCode));
   }
+
+
+  private checkLicenseIfDiscontinued() {
+    if (this.collegeCode.value && this.licenseCode.value) {
+      this.licenseClassDiscontinued = this.isCertificationDiscontinued(this.collegeCode.value, this.licenseCode.value);
+      this.collegeDiscontinued = this.colleges.some(c => c.code == this.collegeCode.value);
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.ts:420]: There is something not clear about this line ... also should that be a triple equals?

File: prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.ts [hunk 10]
```diff
@@ -382,4 +412,17 @@ export class CollegeCertificationFormComponent implements OnInit {
...
+    }
+  }
+
+  private isCertificationDiscontinued(collegeCode: number, licenseCode: number): boolean {
+    let collegeLicense = this.configService.licenses.find(l => l.code === licenseCode);
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.ts:425]: Rather than `collegeLicense`, should that just be `license`?

File: prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.html [hunk 11]
```diff
@@ -1,11 +1,25 @@
...
+           type="danger"
+           icon="warning"
+           class="mb-4">
+  <ng-container #alertContent
+                class="alert-content">
+    {{ this.collegeDiscontinued ?
+    "Your previous licence class is no longer available, please " :
+    "The name of your college has changed. Please select the new name from the dropdown list for college name, " +
+    "then "}}
+    use the dropdown menu to select the licence you currently hold.
```
Comment (neophyte57) [inline prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.html:11]: This seems reversed, possibly because of how it is set in `checkLicenseIfDiscontinued`?

File: prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.ts (hunk 9 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.ts]: Comment added and changed to a triple equals. Also, modified the logic a bit for your comment below.

File: prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.ts (hunk 10 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.ts]: It is updated.

File: prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.html (hunk 11 above)
Comment (bergomi02) [reply prime-angular-frontend/src/app/shared/components/forms/college-certification-form/college-certification-form.component.html]: Good catch. The logic and the code have been updated accordingly.

**Review by neophyte57 — APPROVED** [review]
Well done!  💯
